dockstore-common/src/main/java/io/dockstore/common/yaml/DockstoreYamlHelper.java

```diff
@@ -373,40 +382,82 @@ private static void addNewPropertyClassToQueue(String dockstoreYamlPackageName,
         }
     }
 
-    private static <T> void validate(final T validatee) throws DockstoreYamlException {
+    public static <T> void validate(final T validatee) throws DockstoreYamlException {
+        validate(validatee, true);
+    }
+
+    public static <T> void validate(final T validatee, final boolean validateEntries) throws DockstoreYamlException {
+        validate(validatee, validateEntries, ".dockstore.yml");
+    }
+
+    /**
+     * Performs constraint validation on an object, throwing an exception containing a descriptive error message if there violations.
+     * @param validatee object to validate
+     * @param validateEntries determines whether to validate the entries in a validatee representing a .dockstore.yml, should be false for other types of validatee
+     * @param validateeDescription a very short description of the validatee, used to form error messages for top-level violations
```

Comment:
is the extra e intentional to signal something (real question)? 
If so, maybe it should be in the Javadoc

Reply:
(This function is a variant of an pre-existing function that takes an argument named `validatee`.)

Reply:
I'm ok with the name, maybe just put it in the javadoc since it kinda looks like a typo

Reply:
The `validatee` parameter is already documented in the javadoc, so I'm not sure what you mean.   I can change the `validatee` verbiage in the parameter descriptions if that's what you mean...

Reply:
"Validatee is not a typo and is not a mis-spelling, we're intentionally spelling it this way to denote that validatee is what we're validating" kinda thing
  
https://www.spellcheck.net/misspelled-words/validatee

Reply:
I'll just change the `validatee`-based parameter names to something else and eliminate the issue.

Reply:
Changed `validatee` to `target`.